src/components/BaseSiteBarPlot.jsx

```diff
@@ -0,0 +1,123 @@
+import React from "react";
+const d3 = require("d3");
+const $ = require("jquery");
+
+class BaseSiteBarPlot extends React.Component {
+  constructor(props) {
+    super(props);
+
+    // d3 variables used by multiple methods
+    this.padding_bottom = 15;
+    this.chart_height = this.props.height - this.padding_bottom;
+
+    $("#alignmentjs-siteBarPlot-div").scrollLeft(this.props.x_pixel);
+  }
+
+  componentDidMount() {
+    if (this.props.data != null) {
+      this.createBarPlot();
+    }
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    document
+      .getElementById("alignmentjs-siteBarPlot-div")
+      .addEventListener("alignmentjs_wheel_event", function(e) {
+        $("#alignmentjs-siteBarPlot-div").scrollLeft(e.detail.x_pixel);
+      });
+    if (prevProps.data == null) {
+      this.createBarPlot();
+    } else {
+      this.transitionBarPlot();
+    }
+  }
+
+  createBarPlot() {
+    var data = this.props.data;
+    var chart_height = this.chart_height;
+    var max_value = this.props.max_value && d3.max(data);
+
+    var barChartScale = d3
+      .scaleLinear()
+      .domain([max_value, 0])
+      .range([0, chart_height]);
+
+    var barWidth = this.props.siteSize;
+
+    var chart = d3
+      .select("#alignmentjs-siteBarPlot")
+      .attr("width", this.props.width)
+      .attr("height", chart_height + this.padding_bottom);
+
+    chart
+      .selectAll("g")
+      .data(data)
+      .enter()
+      .append("g")
+      .attr("transform", function(d, i) {
+        return "translate(" + i * barWidth + ", 10)";
```

Comment:
`translate(" + (i * barWidth + 1) + ", 10)` would center the bars. Adding `1` since it is half of the offset to the width used below on line 68 below. I've found that it helps my future self to at the very least declare these as named variables, instead of having a bunch of random numbers floating around. Also, why is this 10 here? I see that is necessary but cannot understand why.

Reply:
Good comment! I'm using named variables to make things clearer. 10 is for the top padding and has been named as such.